### PR TITLE
tfsec to Trivy scanning 

### PR DESCRIPTION
**Pull request by shanice-skylight**
# DEVOPS PULL REQUEST

## Related Issue

- Per warnings found in the most recent executions of tfsec, the Aqua team has shifted their focus to Trivy. tfsec will continue to be available, but it is unclear just how often it will be kept updated.

## Changes Proposed

- Run Trivy as a check on each PR that gets opened in the repository. This helps identify potential vulnerabilities that might get introduced with each PR.
- Disable the tfsec workflow
- Trivy picked up a [misconfiguration](https://github.com/CDCgov/prime-simplereport/actions/runs/9898525226/job/27345533159) in the Dockerfile that it evaluated as High. This check will not pass HIGH or CRITICAL findings are found. So in order to fix it, I created a user called dockeruser, that is used to run the commands.
    - Here's the [reference](https://avd.aquasec.com/misconfig/ds002) for the misconfiguration 

## Additional Information

-This update will run Trivy in Repo Mode: This tells Trivy to scan the repository.
- It will scan for Vulnerabilities, Misconfigurations, and Secrets

## Testing

- Verify trivy scanning results are displayed in the console
- This PR has already been verified on `dev6`

### CODE REVIEW

File: .github/workflows/trivy.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: trivy
+
+on:
+  pull_request:
```
Comment (mehansen) [inline .github/workflows/trivy.yml:4]: do we want this to run on merge to main too?

File: .github/workflows/trivy.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@0.24.0
+        with:
+          scan-type: 'fs'
+          scan-ref: 'ops/'
```
Comment (emyl3) [inline .github/workflows/trivy.yml:24]: is `scan-ref` indicating the directory of our project we are scanning? (based on my understanding of the [docs here](https://github.com/aquasecurity/trivy-action?tab=readme-ov-file#inputs)) -- do we only want to scan the `ops/` repo or the entire project?

File: .github/workflows/trivy.yml (hunk 1 above)
Comment (shanice-skylight) [reply .github/workflows/trivy.yml:4]: Good catch, updated it to run also on a push to main

File: .github/workflows/trivy.yml (hunk 2 above)
Comment (emyl3) [reply .github/workflows/trivy.yml:27]: Got it!! Thank you for explaining! :D

**Review by emyl3 — APPROVED** [review]
thank you for implementing this! based on what I could understand and seeing a successful trivy run https://github.com/CDCgov/prime-simplereport/actions/runs/10097379140/job/27922083672?pr=7905 -- this looks good to me!

**Review by mehansen — APPROVED** [review]

**Comment by mehansen** [comment]
lgtm as well. also checked that the db-client for `dev6` looks to be running properly

**Review by DanielSass — APPROVED** [review]
🥇

**Pull request merged**